sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosPatch.java

```diff
@@ -0,0 +1,198 @@
+     *  3. Target location specifies an object member that does exist, that member's value is replaced.
+     *
+     * @param <T> The type of item to be added.
+     *
+     * @param path the operation path.
+     * @param value the value which will be added.
+     *
+     * @return same instance of {@link CosmosPatch}
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public <T> CosmosPatch add(String path, T value) {
+
+        checkNotNull(value, "expected non-null value");
+        checkArgument(StringUtils.isNotEmpty(path), "path empty %s", path);
+
+        this.patchOperations.add(
+            new PatchOperationCore<>(
+                PatchOperationType.ADD,
+                path,
+                value));
+
+        return this;
+    }
+
+    /**
+     * This removes the value at the target location.
+     *
+     * @param path the operation path.
+     *
+     * @return same instance of {@link CosmosPatch}
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public CosmosPatch remove(String path) {
+
+        checkArgument(StringUtils.isNotEmpty(path), "path empty %s", path);
+
+        this.patchOperations.add(
+            new PatchOperationCore<>(
+                PatchOperationType.REMOVE,
+                path,
+                null));
+
+        return this;
+    }
+
+    /**
+     * This replaces the value at the target location with a new value.
+     *
+     * @param <T> The type of item to be replaced.
+     *
+     * @param path the operation path.
+     * @param value the value which will be replaced.
+     *
+     * @return same instance of {@link CosmosPatch}
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public <T> CosmosPatch replace(String path, T value) {
+
+        checkArgument(StringUtils.isNotEmpty(path), "path empty %s", path);
+
+        this.patchOperations.add(
+            new PatchOperationCore<>(
+                PatchOperationType.REPLACE,
+                path,
+                value));
+
+        return this;
+    }
+
+    /**
+     * This sets the value at the target location with a new value.
+     *
+     * @param <T> The type of item to be set.
+     *
+     * @param path the operation path.
+     * @param value the value which will be set.
+     *
+     * @return same instance of {@link CosmosPatch}
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public <T> CosmosPatch set(String path, T value) {
+
+        checkNotNull(value, "expected non-null value");
+        checkArgument(StringUtils.isNotEmpty(path), "path empty %s", path);
+
+        this.patchOperations.add(
+            new PatchOperationCore<>(
+                PatchOperationType.SET,
+                path,
+                value));
+
+        return this;
+    }
+
+    /**
+     * This increment the value at the target location.
+     *
+     * @param path the operation path.
+     * @param value the value which will be incremented.
+     *
+     * @return same instance of {@link CosmosPatch}
```

Comment:
IMO it would be good to add some comments around expectations for idempotency. Most of the patch operations above should be safely retryable without major side effects - like an add is not technically idempotent because the retry might return a 409 conflict instead of 201-created - but the possible side-effects due to retries are low (no data "corruption") and can be dealt with. For Increment this is not the case - same as replace when not combined with some condition (etag etc. that prevents changes if the property was changed in the meantime).   I raised some concerns around exposing the Increment in the public surface area before - mostly because I think the semantic isn't very clear and it can easily lead to confusion with customers - only way to minimize the risk is by very explicitly calling out the semantics (idempotency, transactional guarantees, consistency guarantees for concurrent incrmenets etc.) and explaining for which scenarios this is useful - and which scenarios cannot be met with it.